app/components/modals/motion_modal.html.erb

```diff
@@ -0,0 +1,30 @@
+<div class="modal-open" data-motion="dismiss" data-value="dismissable">
+  <!-- This modal is styled with Bootstrap but triggered/displayed with motion -->
+  <!-- Bootstrap adds `.modal-open` to body, and instead we create a surrounding div with that class -->
+  <!-- Bootstrap uses css animations that are triggered when `.show` is added after the modal is in the DOM, so this component does that manually -->
+  <!-- Bootstrap sets the style to display: block; (normally display: none;) -->
+  <!-- Instead of using Bootstrap data-dismiss="modal" attributes, we use motion (data-motion="dismiss") to close the modal  -->
+  <div class="modal fade <%= modal_show_class %>" role="dialog" aria-labelledby="motionModalLabel" style="<%= modal_style %>" data-value="dismissable">
+    <div class="modal-dialog" role="document">
+      <div class="modal-content">
+        <div class="modal-header">
+          <h5 class="modal-title" id="motionModalLabel">Motion-Controlled Modal</h5>
+          <!-- Motion is used to dismiss, instead of Bootstrap -->
+          <button type="button" class="close" data-value="dismissable" aria-label="Close">
```

Comment:
this says motion is used, but there's no `data-motion` ? maybe i'm missing something

Reply:
The motion is on the div on line 1.

That's because we want multiple elements in here to trigger motion.  And we'll know if it's an element that should dismiss the modal based on the `data-value` that is on the element.

Reply:
ah yeah, this makes sense now. i guess i was under the impression that each element that could create events needed a `data-motion`. but i guess this goes back to the DOM and how events bubble up to the parent? 

would this be appropriate for something where the modal handled the buttons in a different way? for example, maybe the "Save" button would progress to a second view instead of dismissing, you could put a different `data-value` in and then the Motion component could handle it differently.

Reply:
Yes, you could definitely have the save button do one action, cancel another action, and then dismiss the modal as something common to both.

I used event bubbling is so that clicking outside of the modal also worked to dismiss the modal the way you would expect.